specification/frontdoor/resource-manager/Microsoft.Network/stable/2021-06-01/frontdoor.json

```diff
@@ -0,0 +1,2368 @@
+          "type": "array",
+          "items": {
+            "description": "A load balancing settings",
+            "$ref": "#/definitions/LoadBalancingSettingsModel"
+          }
+        },
+        "healthProbeSettings": {
+          "description": "Health probe settings associated with this Front Door instance.",
+          "type": "array",
+          "items": {
+            "description": "A health probe settings",
+            "$ref": "#/definitions/HealthProbeSettingsModel"
+          }
+        },
+        "backendPools": {
+          "description": "Backend pools available to routing rules.",
+          "type": "array",
+          "items": {
+            "description": "A backend pool",
+            "$ref": "#/definitions/BackendPool"
+          }
+        },
+        "frontendEndpoints": {
+          "description": "Frontend endpoints available to routing rules.",
+          "type": "array",
+          "items": {
+            "description": "A frontend endpoint",
+            "$ref": "#/definitions/FrontendEndpoint"
+          }
+        },
+        "backendPoolsSettings": {
+          "description": "Settings for all backendPools",
+          "$ref": "#/definitions/BackendPoolsSettings"
+        },
+        "enabledState": {
+          "description": "Operational status of the Front Door load balancer. Permitted values are 'Enabled' or 'Disabled'",
+          "enum": [
+            "Enabled",
+            "Disabled"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "FrontDoorEnabledState",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "FrontDoorProperties": {
+      "description": "The JSON object that contains the properties required to create an endpoint.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/FrontDoorUpdateParameters"
+        }
+      ],
+      "properties": {
+        "resourceState": {
+          "description": "Resource status of the Front Door.",
+          "$ref": "#/definitions/ResourceState"
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Provisioning state of the Front Door."
+        },
+        "cname": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The host that each frontendEndpoint must CNAME to."
+        },
+        "frontdoorId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The Id of the frontdoor."
+        },
+        "rulesEngines": {
+          "readOnly": true,
+          "description": "Rules Engine Configurations available to routing rules.",
+          "type": "array",
+          "items": {
+            "description": "A rules engine configuration.",
+            "$ref": "#/definitions/RulesEngine"
+          }
+        },
+        "extendedProperties": {
```

Comment:
>extendedProperties [](http://example.com/codeflow?start=9&length=18)

whats the reason for not having a specific schema for this? Who defines these properties - your service or your customers?

Reply:
These are defined by the service but we set some dynamic values. We got to know that this practice is not recommended recently from another PR that goes in tandem with this: https://github.com/Azure/azure-rest-api-specs/pull/18760 
We will have this in mind when we are developing anything further. But for this PR, we would like to keep it this way as our design and implementation are all done.